packages/forms-web-app/__tests__/unit/controllers/eligibility/granted-or-refused-permission.test.js

```diff
@@ -0,0 +1,115 @@
+const { VIEW } = require('../../../../src/lib/views');
+const logger = require('../../../../src/lib/logger');
+const { APPEAL_DOCUMENT } = require('../../../../src/lib/empty-appeal');
+const { mockReq, mockRes } = require('../../mocks');
+
+jest.mock('../../../../src/lib/appeals-api-wrapper');
+jest.mock('../../../../src/lib/logger');
+
+describe('controllers/eligibility/granted-or-refused-permission', () => {
+  let req;
+  let res;
+  let appeal;
+
+  beforeEach(() => {
+    req = mockReq();
+    res = mockRes();
+
+    ({ empty: appeal } = APPEAL_DOCUMENT);
+
+    jest.resetAllMocks();
+  });
+
+  describe('getGrantedOrRefusedPlanningPermission', () => {
+    it('should call the correct template', () => {
+      grantedOrRefusedPlanningPermissionController.getGrantedOrRefusedPermission(req, res);
+
+      expect(res.render).toHaveBeenCalledWith(VIEW.ELIGIBILITY.GRANTED_REFUSED_PERMISSION, {
+        appeal: req.session.appeal,
+      });
+    });
+  });
+
+  describe('postGrantedOrRefusedPlanningPermission', () => {
+    it('should re-render the template with errors if there is any validation error', async () => {
+      const mockRequest = {
+        ...req,
+        body: {
+          'granted-or-refused-permission': null,
+          errors: { a: 'b' },
+          errorSummary: [{ text: 'There were errors here', href: '#' }],
+        },
+      };
+      await grantedOrRefusedPlanningPermissionController.postGrantedOrRefusedPermission(
+        mockRequest,
+        res
+      );
+
+      expect(createOrUpdateAppeal).not.toHaveBeenCalled();
+
+      expect(res.redirect).not.toHaveBeenCalled();
+
+      expect(res.render).toHaveBeenCalledWith(VIEW.ELIGIBILITY.GRANTED_REFUSED_PERMISSION, {
+        appeal: {
+          ...req.session.appeal,
+          eligibility: {
+            ...req.session.appeal.eligibility,
+            planningPermissionStatus: null,
+          },
+        },
+        errorSummary: [{ text: 'There were errors here', href: '#' }],
+        errors: { a: 'b' },
+      });
+    });
+
+    it('should re-render the template with errors if there is any api call error', async () => {
+      const mockRequest = {
+        ...req,
+        body: {},
+      };
+
+      const error = new Error('Cheers');
+      createOrUpdateAppeal.mockImplementation(() => Promise.reject(error));
+
+      await grantedOrRefusedPlanningPermissionController.postGrantedOrRefusedPermission(
+        mockRequest,
+        res
+      );
+
+      expect(res.redirect).not.toHaveBeenCalled();
+
+      expect(logger.error).toHaveBeenCalledWith(error);
+
+      expect(res.render).toHaveBeenCalledWith(VIEW.ELIGIBILITY.GRANTED_REFUSED_PERMISSION, {
+        appeal: req.session.appeal,
+        errors: {},
+        errorSummary: [{ text: error.toString(), href: '#' }],
+      });
+    });
+
+    it(`'should redirect to '/${VIEW.ELIGIBILITY.DECISION_DATE}' if 'planningPermissionStatus' is 'refused'`, async () => {
```

Comment:
What should happen if the user has selected one of the other 2 options (granted or nodecisionreceived)? 

If it's a different action then perhaps we need tests for that, if it's the same then perhaps we need to add that to teh test description to make it clearer.

Reply:
Those two scenarios will be covered when these cards - [AS-3291](https://pins-ds.atlassian.net/browse/AS-3291) and [AS-3292](https://pins-ds.atlassian.net/browse/AS-3292) - are played